src/lib/Input.tsx

```diff
@@ -27,6 +27,27 @@ const Input = <T extends FieldValues>(props: InputProps<T>) => {
   if (props.type === "select" && !props.options) {
     throw new Error("options must be provided for select inputs");
   }
+  if (
```

Comment:
I would suggest to make a const list with all the allowed types to have an icon and than in the if simply check if the actual type is included in the list of allowed types. Makes it a bit more readable.